### ISSUE TITLE
False positive: "Objects are always passed by reference" with mixed type

### ISSUE THREAD

**Issue by voku** [kalessil/phpinspectionsea#1668]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Plugin         | Php Inspections (EA Ultimate), 3.0.10 |
| Language level | PHP 8.0                 |

Current behaviour
--
```
/**
 * Binds variables to a prepared statement for result storage
 * @link https://php.net/manual/en/mysqli-stmt.bind-result.php
 * @param mysqli_stmt $statement Statement
 * @param mixed &$var1 The variable to be bound.
 * @param mixed &...$_ The variables to be bound.
 * @return bool
 */
function mysqli_stmt_bind_result (mysqli_stmt $statement, mixed &$var1, &...$_): bool
{}
```

> [EA] Objects are always passed by reference; please correct '& $var1'. 

Expected behaviour
--
no error

Environment details
---
PhpStorm 2020.3
Build #PS-203.5981.175, built on December 2, 2020
Licensed to Portable UTF-8 / Lars Moelleken
Subscription is active until February 27, 2021
For non-commercial open source development only.
Runtime version: 11.0.9+11-b1145.21 amd64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
Linux 4.15.0-126-generic
GC: G1 Young Generation, G1 Old Generation
Memory: 5120M
Cores: 4
Registry: run.processes.with.pty=TRUE, ide.tooltip.initialDelay=885, ide.balloon.shadow.size=0
Non-Bundled Plugins: DBN, Key Promoter X, String Manipulation, awesome.console, com.alayouni.ansiHighlight, com.andrey4623.rainbowcsv, com.intellij.ideolog, dev.arunvelsriram.desccron, net.seesharpsoft.intellij.plugins.csv, uk.co.ben-gibson.remote.repository.mapper, com.jetbrains.codeWithMe, name.kropp.intellij.makefile, NEON support, com.kalessil.phpStorm.phpInspectionsUltimate, me.artspb.idea.eval.plugin, net.king2500.plugins.PhpAdvancedAutoComplete, lv.midiana.misc.phpstorm-plugins.deep-keys, de.espend.idea.php.toolbox, de.espend.idea.php.annotation, fr.adrienbrault.idea.symfony2plugin, izhangzhihao.rainbow.brackets, ru.adelf.idea.dotenv
Current Desktop: ubuntu:GNOME

**Comment by ea-inspections-team** [comment]
Thanks for reporting @voku! Fixed!
